app/views/shared/_orders.html.erb

```diff
@@ -0,0 +1,84 @@
+        <th> Order Date </th>
+        <% if current_admin? %>
+          <th class="dropdown">
+             <a href="#" data-toggle="dropdown" class="dropdown-toggle">Order Status <b class="caret"></b></a>
+             <ul class="dropdown-menu">
+               <% if @status_filter %>
+                  <li> <%= link_to "Clear filter", admin_dashboard_path(status_filter: nil), class: "dropdown-option" %>
+                  <li role="separator" class="divider"></li>
+               <% end %>
+               <% @statuses.each do |status| %>
+                  <li> <%= link_to "#{status.name.capitalize}", admin_dashboard_path(status_filter: status.name), class: "dropdown-option" %> </li>
+               <% end %>
+             </ul>
+          </th>
+        <% else %>
+          <th class="dropdown">
+             <a href="#" data-toggle="dropdown" class="dropdown-toggle">Order Status <b class="caret"></b></a>
+             <ul class="dropdown-menu">
+               <% if @status_filter %>
+                  <li> <%= link_to "Clear filter", orders_path(status_filter: nil), class: "dropdown-option" %>
+                  <li role="separator" class="divider"></li>
+               <% end %>
+               <% @statuses.each do |status| %>
+                  <li> <%= link_to "#{status.name.capitalize}", orders_path(status_filter: status.name), class: "dropdown-option" %> </li>
+               <% end %>
+             </ul>
+          </th>
+        <% end %>
+        <% if current_admin? %>
+          <th> Mark as completed </th>
+          <th> Mark as paid </th>
+          <th> Cancel </th>
+        <% end %>
+      </tr>
+    </thead>
+    <tbody>
+      <% @orders.each do |order| %>
+      <tr class= "order_<%= order.id %>_row">
+        <td> <%= link_to "#{order.id}", admin_order_path(order), class: "order_#{order.id}" %> </td>
+        <% if current_admin? %>
+          <td> <%= order.customer.id %> </td>
+          <td> <%= order.customer.full_name %> </td>
+          <td> <%= order.customer.email %> </td>
+        <% end %>
+        <td> <%= number_to_currency(order.total_price_in_dollars) %> </td>
+        <td> <%= order.date %> </td>
+        <td>
+          <span class="status_<%= order.status_name.downcase %>">
+            <%= order.status.name.capitalize %>
+          </span>
+        </td>
+        <% if current_admin? %>
+        <td>
+          <% if order.status.paid? %>
+            <%= button_to "Mark as completed", orders_path, method: :put, class: "order_#{order.id}_complete", params: {"order_id" => order.id, "status" => "completed"} %>
+          <% end %>
+        </td>
+        <td>
+          <% if order.status.name == "ordered" %>
+            <%= button_to "Mark as paid", orders_path, method: :put, class: "order_#{order.id}_mark_as_paid", params: {"order_id" => order.id, "status" => "paid"} %>
+          <% end %>
+        </td>
+        <td>
+          <% if order.status.paid? || order.status.ordered? %>
```

Comment:
this is also a lot of logic in your view. can we create a method in `Order` that does this for us? maybe something called `paid_or_ordered?`?